File: trees/Binary Tree/construction.py
import numpy as np

class node:
	def __init__(self,data):
		self.left = None
		self.right = None
		self.data = data

def display(head):
	if head == None:
		return
	display(head.left)
	print head.data
	display(head.right)


head = node(10)
head.left = node(5)
head.right = node(14)
head.right.left = node(12)
print "hello"
display(head)

File: trees/Binary Tree/construction.cpp
#include<iostream>
using namespace std;

class node {
public:
	node* left;
	node* right;
	int data;
};

node* createnode(int data) {
	node* newnode = new node;
	newnode->left = NULL;
	newnode->right = NULL;
	newnode->data = data;
	return newnode;
}

void insert(node*& head, int data) {
	if(head == NULL) {
		head = createnode(data);
	}
	else {
		if(head->data >= data){
			insert(head->left, data);
		}
		else {
			insert(head->right, data);
		}
	}
}

void display(node* head) {
	if(head == NULL)
		return;
	display(head->left);
	cout<<head->data<<" ";
	display(head->right);
}

int main() {
	node* head = NULL;
	insert(head, 10);
	insert(head, 5);
	insert(head, 14);
	insert(head, 12);
	display(head);
	return 0;
}